app/views/questionnaire.py

```diff
@@ -199,6 +215,50 @@ def get_thank_you():
     return _redirect_to_latest_location(routing_path, collection_id, metadata.get('eq_id'), metadata.get('form_type'))
 
 
+@post_submission_blueprint.route('/view-submission', methods=['GET'])
+@login_required
+def get_view_submission():  # pylint: disable=unused-argument
+
+    session_data = get_session_store().session_data
+    submission_valid_until = int(dateutil.parser.parse(session_data.submitted_time).strftime('%s')) \
+        + session_data.submission_validity
+
+    if submission_valid_until > int(datetime.utcnow().strftime('%s')):
+
+        submitted_data = SubmittedResponses().get_item(session_data.tx_id)
+
+        if submitted_data and submitted_data['valid_until'] > int(datetime.utcnow().strftime('%s')):
+
+            metadata_context = build_metadata_context_for_survey_completed(vars(session_data))
+
+            current_location = Location('group', 0, 'view-submission')
+            encrypter = StorageEncryption(current_user.user_id, current_user.user_ik)
+
+            answer_store = AnswerStore()
+            submitted_data = json.loads(encrypter.decrypt_data(submitted_data['data']))
+            answer_store.add_multiple_answers(submitted_data.get('answers'))
+
+            g.schema_json = load_schema_from_params(session_data.eq_id, session_data.form_type)
+            metadata = submitted_data.get('metadata')
+
+            routing_path = PathFinder(g.schema_json, answer_store, metadata).get_full_routing_path()
+
+            schema_context = _get_schema_context(routing_path, current_location, metadata, answer_store)
+            rendered_schema = renderer.render(g.schema_json, **schema_context)
+            summary_rendered_context = build_summary_rendering_context(rendered_schema, answer_store, metadata)
+
+            submission_view_template = render_theme_template(g.schema_json['theme'],
+                                                             template_name='view-submission.html',
+                                                             meta=metadata_context,
+                                                             analytics_ua_id=current_app.config['EQ_UA_ID'],
+                                                             survey_id=g.schema_json['survey_id'],
+                                                             survey_title=TemplateRenderer.safe_content(g.schema_json['title']),
+                                                             summary=summary_rendered_context)
+            return submission_view_template
+
+    return get_thank_you()
```

Comment:
I think we should re-direct to the thank you page otherwise you will have the thank you page with the url of `/view-submission`